### ISSUE TITLE
Make label / candidates list to be one metadata of model type

### ISSUE THREAD

**Issue by eddyxu** [eto-ai/rikai#592]
# Problem

Our model type only converts the tensors of label ids into `int` in the ModelType output, which makes the outputs not human readable, unless users convert id to text later by themselves.

# Solution

Model type can specify a callable or an object that returns the string/text representation of label ID  from detections. It also helps users who take off-the-shelf models and train them on new datasets with different labels.

**Comment by changhiskhan** [comment]
@eddyxu based on my understanding, whether it's pretrained or custom trained, we'd still be using the same `ModelType` right? So it would seem like this mapping can't always be a part of the `ModelType`?
I was thinking that we could:
1. Have ModelType specify a default mapping that can be used only when using a pretrained model.
2. Overrides can be specified via the actual model spec for a specific model artifact. 

Override mapping:
1. For yaml spec, the actual mapping can be written as json in a separate file and the yaml spec will contain the url of that file
2. For mlflow spec, the mapping can be written as json in a separate file and we log it as ["extra_files"](https://www.mlflow.org/docs/latest/python_api/mlflow.pytorch.html#mlflow.pytorch.log_model)

We can implement this in multiple tickets, starting with just the pre-trained models.

wdyt?

**Comment by eddyxu** [comment]
It might be similar to `ModelType.pretrain_fn` or `TorchModelType.collate_fn`(TBD) that `ModelType` can offer a property / function / callable or list: i.e., `ModelType.label(label_id: int) -> str`?

In such case, we can let `mlflow.log_model` to register the implementation of this method as metadata / tags.

Also, lets assume that users start to train a model with a completed new set of labels.

they can do

```
from rikai ... import model_type

@model_type(name="my_model", labels=...)
class MyNewModel(torch.nn.Module):
    pass
```

Or 

```
model = torchvision.models.ssd()

mlflow.log_model(model,  labels=...)
```

Not sure how big can be the payload is in `mlflow` tags. Would be nice just store them in place (either a list of labels, or the module name to a callable)?

**Comment by changhiskhan** [comment]
Mlflow tag value is 5000 bytes max. Logodet3K for example has 3000 logo categories.
